src/main/java/net/glowstone/net/codec/play/game/NBTQueryResponseCodec.java

```diff
@@ -0,0 +1,25 @@
+package net.glowstone.net.codec.play.game;
+
+import com.flowpowered.network.Codec;
+import com.flowpowered.network.util.ByteBufUtils;
+import io.netty.buffer.ByteBuf;
+import net.glowstone.net.message.play.game.NBTQueryResponseMessage;
+
+import java.io.IOException;
+
+public final class NBTQueryResponseCodec implements Codec<NBTQueryResponseMessage> {
+
+    @Override
+    public NBTQueryResponseMessage decode(ByteBuf byteBuf) throws IOException {
+        int transactionID = ByteBufUtils.readVarInt(byteBuf);
+        String nbt = ByteBufUtils.readUTF8(byteBuf);
```

Comment:
This should be stored, encoded, and decoded as a `CompoundTag`. In GlowBufUtils there is are functions to encode/decode it (readCompound and writeCompound)